tests/unit/uncertainty/test_evaluators.py

```diff
@@ -1,18 +1,157 @@
+            torch.tensor([0.2]),
+        ),
+    ],
+)
+def test_NLLClassEvaluator(preds, uncs, targets, mask, likelihood):
+    """
+    Testing the NLLClassEvaluator
+    """
+    evaluator = NLLClassEvaluator()
+    nll_calc = evaluator.evaluate(preds, uncs, targets, mask)
+    likelihood_calc = torch.exp(-1 * nll_calc)
+    torch.testing.assert_close(likelihood_calc, likelihood)
+
+
+@pytest.mark.parametrize(
+    "preds,uncs,targets,mask,likelihood",
+    [
+        (
+            torch.tensor(
+                [
+                    [[0.29, 0.22, 0.49]],
+                    [[0.35, 0.19, 0.46]],
+                    [[0.55, 0.38, 0.07]],
+                    [[0.15, 0.29, 0.56]],
+                    [[0.08, 0.68, 0.24]],
+                ]
+            ),
+            torch.tensor(
+                [
+                    [[0.29, 0.22, 0.49]],
+                    [[0.35, 0.19, 0.46]],
+                    [[0.55, 0.38, 0.07]],
+                    [[0.15, 0.29, 0.56]],
+                    [[0.08, 0.68, 0.24]],
+                ]
+            ),
+            torch.tensor([[0], [2], [2], [0], [1]]),
+            torch.ones([5, 1], dtype=bool),
+            torch.tensor([0.24875443]),
+        ),
+        (
+            torch.tensor(
+                [
+                    [[8.7385e-01, 8.3770e-04, 3.3212e-02, 9.2103e-02]],
+                    [[7.2274e-03, 1.0541e-01, 8.8703e-01, 3.2886e-04]],
+                    [[1.7376e-03, 9.9478e-01, 1.4227e-03, 2.0596e-03]],
+                    [[2.6487e-04, 1.3251e-03, 2.4325e-02, 9.7409e-01]],
+                ]
+            ),
+            torch.tensor(
+                [
+                    [[8.7385e-01, 8.3770e-04, 3.3212e-02, 9.2103e-02]],
+                    [[7.2274e-03, 1.0541e-01, 8.8703e-01, 3.2886e-04]],
+                    [[1.7376e-03, 9.9478e-01, 1.4227e-03, 2.0596e-03]],
+                    [[2.6487e-04, 1.3251e-03, 2.4325e-02, 9.7409e-01]],
+                ]
+            ),
+            torch.tensor([[0], [2], [1], [3]]),
+            torch.ones([4, 1], dtype=bool),
+            torch.tensor([0.93094635]),
+        ),
+    ],
+)
+def test_NLLMultiEvaluator(preds, uncs, targets, mask, likelihood):
+    """
+    Testing the NLLMultiEvaluator
+    """
+    evaluator = NLLMultiEvaluator()
+    nll_calc = evaluator.evaluate(preds, uncs, targets, mask)
+    likelihood_calc = torch.exp(-1 * nll_calc)
+    torch.testing.assert_close(likelihood_calc, likelihood)
+
+
+@pytest.mark.parametrize(
+    "preds,uncs,targets,mask,likelihood",
+    [
+        (
+            torch.zeros([1, 1]),
+            torch.ones([1, 1]),
+            torch.zeros([1, 1]),
+            torch.ones([1, 1], dtype=bool),
+            torch.tensor([0.39894228]),
+        ),
+        (
+            torch.zeros([2, 2]),
+            torch.ones([2, 2]),
+            torch.zeros([2, 2]),
+            torch.ones([2, 2], dtype=bool),
+            torch.tensor([0.39894228, 0.39894228]),
+        ),
```

Comment:
same here

Reply:
This is testing `NLLRegressionEvaluator` which does use `preds`, so I have left it unchanged. I believe you meant the `NLLClassEvaluator` test from lines 12 to 38, which I have changed the `preds` to `None`.